mgmt/imgmgr/src/imgmgr_cli.c

```diff
@@ -114,34 +130,62 @@ imgr_cli_hash_parse(const char *arg, int *out_slot)
 
     rc = hex_parse(arg, strlen(arg), hash, sizeof hash);
     if (rc != sizeof hash) {
-        console_printf("Invalid hash: %s\n", arg);
         return SYS_EINVAL;
     }
 
     slot = imgr_find_by_hash(hash, &ver);
     if (slot == -1) {
-        console_printf("Unknown img\n");
         return SYS_ENOENT;
     }
 
     *out_slot = slot;
     return 0;
 }
 
+static int
+imgr_cli_slot_or_hash_parse(const char *arg, int *out_slot)
+{
+    int rc;
+
+    /* First, parse argument as a slot number.  Parts of the system assume slot
+     * is 0 or 1; enforce those bounds here.
+     */
+    *out_slot = parse_ll_bounds(arg, 0, 1, &rc);
+    if (rc == 0) {
```

Comment:
checking `rc == SYS_EOK` would probably be better here in the long run...

Reply:
Thanks, @utzig.  It might just be a pet peeve of mine, but I really don't like named "OK" return codes.  To me, such constants look like errors at first glance, and it takes a second look to understand what is actually happening.

I think most of the code in core just uses `0` to signify success, so if it is OK with you, I would like to keep this as it is.

Reply:
Yup, I prefer `0` as non-error return value as well (unless it's a bool return); but maybe not having `*_EOK` would be better than...